.github/workflows/clp-core-build.yaml

```diff
@@ -96,37 +114,127 @@ jobs:
         run: "chown $(id -u):$(id -g) -R ."
         shell: "bash"
 
-      - name: "Filter relevant changes"
-        uses: "dorny/paths-filter@v2"
-        id: "filter"
+      - uses: "./.github/actions/clp-core-build-containers"
         with:
-          base: "${{env.PATHS_FILTER_BASE}}"
-          filters: |
-            image:
-              - ".github/actions/**"
-              - ".github/workflows/clp-core-build.yaml"
-              - ${{env.PATHS_FILTER_LIB_INSTALL_GLOB}}
-              - "components/core/tools/docker-images/clp-core-${{env.OS_NAME}}/**"
-              - "components/core/tools/docker-images/clp-env-base-${{env.OS_NAME}}/**"
-              - "components/core/tools/scripts/lib_install/${{env.OS_NAME}}/**"
-            ${{inputs.PATHS_FILTER_CLP_FILTER}}
-
-      - uses: "./.github/actions/clp-core-build-deps-and-binaries"
-        with:
-          os_name: "${{env.OS_NAME}}"
-          deps_image_changed: "${{steps.filter.outputs.image}}"
-          push_binaries_image: true
-          local_registry_port: "5000"
+          os_name: "ubuntu-focal"
+          clp_core_dir: "components/core"
+          push_deps_image: >-
+            ${{github.event_name != 'pull_request' && github.ref == 'refs/heads/main'}}
           token: "${{secrets.GITHUB_TOKEN}}"
 
-  ubuntu-jammy:
+  ubuntu-jammy-deps-image:
+    if: "needs.filter-relevant-changes.outputs.ubuntu_jammy_image_changed == 'true'"
+    needs: "filter-relevant-changes"
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: "actions/checkout@v3"
+        with:
+          submodules: "recursive"
+
+      - name: "Work around actions/runner-images/issues/6775"
+        run: "chown $(id -u):$(id -g) -R ."
+        shell: "bash"
+
+      - uses: "./.github/actions/clp-core-build-containers"
+        with:
+          os_name: "ubuntu-jammy"
+          clp_core_dir: "components/core"
+          push_deps_image: >-
+            ${{github.event_name != 'pull_request' && github.ref == 'refs/heads/main'}}
+          token: "${{secrets.GITHUB_TOKEN}}"
+
+  centos74-binaries:
+    # Run if the ancestor jobs succeeded OR they were skipped and clp was changed.
+    if: >-
+      success()
+      || (!cancelled() && !failure() && needs.filter-relevant-changes.outputs.clp_changed == 'true')
+    needs:
+      - "centos74-deps-image"
+      - "filter-relevant-changes"
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: "actions/checkout@v3"
+        with:
+          submodules: "recursive"
+
+      - name: "Work around actions/runner-images/issues/6775"
+        run: "chown $(id -u):$(id -g) -R ."
+        shell: "bash"
+
+      - uses: "./.github/actions/clp-core-build"
+        with:
+          os_name: "centos7.4"
```

Comment:
Should pass as input to clp-core-build the image name based on an output from centos74-deps-image to guarantee it matches

Reply:
Did my best to centralize the definitions of different variables. For the qualified container name, although we could share it between the jobs, it would require us to compute it per OS and then share it which seems like unnecessary code bloat. Also note that GH actions doesn't allow defining env vars in terms of other env vars.